backend/LexBoxApi/GraphQL/ProjectMutations.cs

```diff
@@ -87,6 +96,64 @@ public record CreateProjectResponse(Guid? Id, CreateProjectResult Result);
         return dbContext.Projects.Where(p => p.Id == input.ProjectId);
     }
 
+    [Error<NotFoundException>]
+    [Error<DbError>]
+    [AdminRequired]
+    [UseMutationConvention]
+    public async Task<BulkAddProjectMembersResult> BulkAddProjectMembers(
+        LoggedInContext loggedInContext,
+        BulkAddProjectMembersInput input,
+        LexBoxDbContext dbContext)
+    {
+        var admin = await dbContext.Users.FindAsync(loggedInContext.User.Id);
+        var project = await dbContext.Projects.FindAsync(input.ProjectId);
+        if (project is null) throw new NotFoundException("Project not found");
+        List<string> usernameConflicts = [];
+        int createdCount = 0;
+        foreach (var username in input.Usernames)
+        {
+            var user = await dbContext.Users.Where(u => u.Username == username).FirstOrDefaultAsync();
+            if (user is null)
+            {
+                createdCount++;
+                var salt = Convert.ToHexString(RandomNumberGenerator.GetBytes(SHA1.HashSizeInBytes));
+                var isEmailAddress = username.Contains('@');
+                // TODO: In the future we'll want to allow usernames in the form "Real Name <email@example.com>" and extract the real name from them
+                // For now, just:
+                var name = username;
+                user = new User
+                {
+                    Id = Guid.NewGuid(),
+                    Username = isEmailAddress ? null : username,
+                    Name = name,
+                    Email = isEmailAddress ? username : null,
+                    LocalizationCode = "en", // TODO: input.Locale,
+                    Salt = salt,
+                    PasswordHash = PasswordHashing.HashPassword(input.PasswordHash, salt, true),
+                    IsAdmin = false,
+                    EmailVerified = false,
+                    CreatedBy = admin,
```

Comment:
instead of querying out the admin we can just set `CreatedById` here, then eliminate the query above where we get the admin from the db.

Reply:
Already done earlier thanks to Tim suggesting it.